pkg/templating/template.go

```diff
@@ -0,0 +1,114 @@
+package templating
+
+import (
+	"fmt"
+	"html/template"
+	"strings"
+
+	"github.com/Masterminds/sprig/v3"
+)
+
+// The split functions cannot be replaced with a single generic one, because there is
+// no common type between maps and slices
+
+// Given a map `src`, return two maps where the first contains key-value pairs that are not
+// templates, and the latter containers only key-value pairs that are templates.
+func SplitTemplatedMapItems(src map[string]string) (map[string]string, map[string]string) {
+	nonTemplateKeyPairs := make(map[string]string, len(src))
```

Comment:
I think the size of the map can be omitted here?

Reply:
It can be, but it's generally faster to specify the size if it's known ahead of time. This avoids overallocation (often by a significant amount) later, or resizing.